### PR TITLE
Help g++ to find the right version of next()

### PR DESCRIPTION
**Pull request by afabri**
With the `Exact_predicates_exact_constructions_kernel` included gcc4.4 has a problem with ADL for for function `next`.  To help it we  provide the types needed for `std::iterator_traits<Surface_mesh::halfedge_index>`, although this descriptor is not an iterator. 

This should fix this [test](https://cgal.geometryfactory.com/CGAL/Members/testsuite/CGAL-4.8-Ic-116/Surface_mesh_Examples/TestReport_lrineau_CentOS6-CXX11-Boost157.gz)

### CODE REVIEW

**Comment by sloriot** [comment]
Merged in CGAL-4.8-Ic-117

**Comment by sloriot** [comment]
Errors are gone

File: Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h [hunk 1]
```diff
@@ -149,6 +149,12 @@ namespace CGAL {
     {
     public:
 
+        typedef void iterator_category;
+        typedef void value_type;
+        typedef void difference_type;
+        typedef void pointer;
+        typedef void reference;
+
         SM_Halfedge_index() : SM_Index<SM_Halfedge_index>(-1) {}
```
Comment (lrineau) [inline Surface_mesh/include/CGAL/Surface_mesh/Surface_mesh.h]: As I said, that patch cannot be understood without comments!

**Comment by sloriot** [comment]
Fixed by f5aae4e

**Pull request merged**